packages/components/src/menu/types.ts

```diff
@@ -173,73 +197,137 @@ export interface MenuItemProps {
 	 */
 	render?: Ariakit.MenuItemProps[ 'render' ];
 	/**
-	 * The ariakit store. This prop is only meant for internal use.
+	 * The ariakit menu store. This prop is only meant for internal use.
 	 * @ignore
 	 */
 	store?: Ariakit.MenuItemProps[ 'store' ];
 }
 
-export interface MenuCheckboxItemProps
-	extends Omit< MenuItemProps, 'prefix' | 'hideOnClick' > {
```

Comment:
Removed the `extends` syntax so that each component uses the correct ariakit types (ie. `Menu.Item` uses `Ariakit.MenuItemProps`, `Menu.CheckboxItem` uses `Ariakit.MenuItemCheckboxProps`, and `Menu.RadioItem` uses `Ariakit.MenuItemRadioProps`